hudi-flink/src/main/java/org/apache/hudi/util/StreamerUtil.java

```diff
@@ -242,7 +242,9 @@ public static TypedProperties flinkConf2TypedProperties(Configuration conf) {
         properties.put(option.key(), option.defaultValue());
       }
     }
-    return new TypedProperties(properties);
+    TypedProperties typeProps = new TypedProperties();
+    typeProps.putAll(typeProps);
```

Comment:
hi, @hk-lrzy thanks for your contribution.
There might be a mistake here , do you mean ` typeProps.putAll(properties)`?

BTW, I think it's better to fix this problem in the constructor of `TypedProperties `, that's where the problem really exists.
WDYT?

Reply:
Yeah，you are right， it should be `typeProps.putAll(properties)`.
About question:

Reply:
> hi @wangxianghu , Yeah，you are right， it should be `typeProps.putAll(properties)`.
> I fix this problems cause by the code
> 
> ```
>     public Builder withProps(Map kvprops) {
>       props.putAll(kvprops);
>       return this;
>     }
> ```
> 
> which invoked by `StreamerUtil`.`getHoodieClientConfig`,
> `withProps` use `putAll` method which only in interface `Map`，but if we create `TypeProperties` use
> 
> ```
>   public TypedProperties(Properties defaults) {
>     super(defaults);
>   }
> ```
> 
> the `kvprops` will be empty because the actually value contained in the field `defaults`.

Sorry, I can understand what you mean.  can you paste the exception you met? 
The test below passed in my local. 
```
  @Test
  public void TestTypedProperties () {
    Properties prop = new Properties();
    prop.put("hudi","hoodie");

    TypedProperties typedProperties = new TypedProperties(prop);
    Assert.assertEquals("hoodie", typedProperties.getString("hudi"));

    typedProperties.put("hudi","hudi");
    Assert.assertEquals("hudi", typedProperties.getString("hudi"));
  }
```

Reply:
```
  @Test
  public void testTypedProperties() throws NoSuchFieldException, IllegalAccessException {
    Properties prop = new Properties();
    prop.put("hudi_key_1", "hoodie");

    TypedProperties typedProperties = new TypedProperties(prop);
    Assert.assertEquals("hoodie", typedProperties.getString("hudi_key_1"));

    typedProperties.put("hudi_key_2", "hoodie");
    Assert.assertEquals("hoodie", typedProperties.getString("hudi_key_2"));

    HoodieWriteConfig.Builder builder = HoodieWriteConfig.newBuilder()
      .withProps(typedProperties);

    Field props = HoodieWriteConfig.Builder.class.getDeclaredField("props");
    props.setAccessible(true);

    String hudi_key_1 = ((Properties) props.get(builder)).getProperty("hudi_key_1");
    Assert.assertNull(hudi_key_1);
    String hudi_key_2 = ((Properties) props.get(builder)).getProperty("hudi_key_2");
    Assert.assertEquals("hoodie", (hudi_key_2));

  }
```

Reply:
> @wangxianghu
> 
> ```
>   @Test
>   public void testTypedProperties() throws NoSuchFieldException, IllegalAccessException {
>     Properties prop = new Properties();
>     prop.put("hudi_key_1", "hoodie");
> 
>     TypedProperties typedProperties = new TypedProperties(prop);
>     Assert.assertEquals("hoodie", typedProperties.getString("hudi_key_1"));
> 
>     typedProperties.put("hudi_key_2", "hoodie");
>     Assert.assertEquals("hoodie", typedProperties.getString("hudi_key_2"));
> 
>     HoodieWriteConfig.Builder builder = HoodieWriteConfig.newBuilder()
>       .withProps(typedProperties);
> 
>     Field props = HoodieWriteConfig.Builder.class.getDeclaredField("props");
>     props.setAccessible(true);
> 
>     String hudi_key_1 = ((Properties) props.get(builder)).getProperty("hudi_key_1");
>     Assert.assertNull(hudi_key_1);
>     String hudi_key_2 = ((Properties) props.get(builder)).getProperty("hudi_key_2");
>     Assert.assertEquals("hoodie", (hudi_key_2));
> 
>   }
> ```

I get your point, values stored in different places. is this a problem in your scenes？ 
I am not sure if this change is necessary, since very few people would use reflection to get the values.

Reply:
@wangxianghu 
Use reflection just want to show the problem, method `StreamerUtil.getHoodieClientConfig` used `withProps` and it caused property value lose.

Reply:
@can you fix the problem in `HoodieWriteConfig`, this bug belongs `org.apache.hudi.config.HoodieWriteConfig.Builder#withProps` right？

Reply:
I think the reason for this bug is the `typeproperties` we used incorrectly.

Reply:
> I think the reason for this bug is the `typeproperties` we used incorrectly.

I think we should strive to make our code more robust, instead of requiring users to use it in a specified way.